src/modules/client.ts

```diff
@@ -91,7 +91,7 @@ export default class Client {
     return {
       fetchOptions:
         typeof this.fetchOptions === 'function'
-          ? this.fetchOptions()
+          ? this.fetchOptions({ headers: defaultHeaders })
```

Comment:
Let's maybe move this into `http-exchange` so that _if_ we want to go for defaults as an argument we can reconstruct the object every time and don't import `defaultHeaders` here as indicated in the next comment

Reply:
Given statements above it makes sense.